### DISCUSSION TITLE
Add support for building with a native TLS library instead of rustls

### DISCUSSION THREAD

**Discussion by jirutka** [meilisearch/product#645] [Feedback & Feature Proposal]
Can you please add support for native TLS library, i.e. allow linking against system-provided TLS library?

Rustls can't be built as a shared library, so it's always statically linked (bundled) into the application binary. So instead of having one system TLS library that can be independently patched for a vulnerability, you end up with who-knows-how-many binaries with different versions of the TLS library bundled inside.

It doesn't matter if the library is written in C or Rust, it is still prone to various types of bugs, and in the case of a crypto library, they can lead to critical vulnerabilities. Also, `rustls` is not written purely in Rust, it is built on top of [ring](https://github.com/briansmith/ring), which is written in assembly and C. Another problem is that `ring` only supports a limited set of CPU architectures (e.g. ppc64le is not supported).

Moreover, `actix-tls` brings dependency on [webpki-roots](https://crates.io/crates/webpki-roots) with the `rustls` feature flag, so it embeds a whole bundle of root CAs into the binary! This is very bad, both from the security perspective and user experience, and unacceptable for Linux distributions.

**Comment by dureuill** [comment]
Hello @jirutka :wave:

We discussed this internally with the team, and we're not ready to commit to adding a native TLS library option to Meilisearch.

openSSL has historically been a source of support issues for us, especially in docker and macOS environments, and we're currently very busy with the upcoming 1.2 release. 

We might consider adding the option as a `meilisearch` "`openssl`" feature (enabling `actix-http`'s `openssl` feature instead of `rustls`) in the future.

Thank you for the well-motivated feature request ❤️
